### PR TITLE
Nukes mutant bodyparts entirely

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Mutant bodyparts list is dead, the last thing that used it (ears) have been moved to a bodypart overlay

This was a bit cursed - Ears themselves are internal organs but all of the bodypart overlay api is on external organ. 
I kinda walked the middle of the line, allowing ears to define a bodypart overlay. This means you can't restyle cat ears, but I'm sure someone can work on that in the future (considering you already cant (afaik))

Other things -

Legs were entirely killed from mutant bodyparts as well, even though there was pretty much no handling for it there anyways. I took the opportunity to fix a bug involving prefs menu not showing digi legs correctly. 

## Why It's Good For The Game

No more this

image

## Changelog

:cl: Melbert
refactor: Cat ears refactored in their entirety. Report any oddities
fix: Changing digi legs in the prefs menu actually updates the dummy without necessitating you re-open it
/:cl:

### CODE REVIEW

File: code/datums/dna.dm [hunk 1]
```diff
@@ -672,6 +672,8 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 		dna.features["tail_monkey"] = SSaccessories.tails_list_monkey[deconstruct_block(get_uni_feature_block(features, DNA_MONKEY_TAIL_BLOCK), length(SSaccessories.tails_list_monkey))]
 	if(dna.features["ears"])
 		dna.features["ears"] = SSaccessories.ears_list[deconstruct_block(get_uni_feature_block(features, DNA_EARS_BLOCK), length(SSaccessories.ears_list))]
+		var/obj/item/organ/internal/ears/ears = get_organ_by_type(__IMPLIED_TYPE__)
+		ears?.imprint_ear(src)
```
Comment (MrMelbert) [inline code/datums/dna.dm:676]: I haven't tested genetics yet but we don't have a super good API for reacting to certain DNA blocks changing

File: code/modules/client/preferences/_preference.dm [hunk 2]
```diff
@@ -336,8 +332,7 @@ GLOBAL_LIST_INIT(preference_entries_by_key, init_preference_entries_by_key())
 	SHOULD_NOT_SLEEP(TRUE)
 
 	if ( \
-		!isnull(relevant_mutant_bodypart) \
-		|| !isnull(relevant_inherent_trait) \
+		!isnull(relevant_inherent_trait) \
 		|| !isnull(relevant_external_organ) \
 		|| !isnull(relevant_head_flag) \
 		|| !isnull(relevant_body_markings) \
```
Comment (ZephyrTFA) [inline code/modules/client/preferences/_preference.dm:338]: :confounded:

File: code/datums/sprite_accessories.dm [hunk 3]
```diff
@@ -1914,11 +1905,12 @@
 	icon_state = "round"
 
 /datum/sprite_accessory/ears/fox
+	// This shouldn't evem be in the cat ear accessory list at all,
+	// but bodypart overlays don't support that yet
```
Comment (MrMelbert) [inline code/datums/sprite_accessories.dm:1909]: CC @Time-Green 
I'd like to make the `ears_list` JUST a list of cat ears, but in testing we don't seem to have a good way to make `/datum/bodypart_overlay/mutant`s for a single sprite overlay
(IE: `/datum/bodypart_overlay/mutant/ears/fox` which JUST does `/datum/sprite_accessory/ears/fox`)
Am I missing something?

**Comment by MrMelbert** [comment]
Ugh I gotta add support for `RESET_COLOR` to GFI to make the tests pass

File: code/datums/sprite_accessories.dm (hunk 3 above)
Comment (Time-Green) [reply code/datums/sprite_accessories.dm:1909]: You can set sprite_datum and then set imprint_on_next_insertion to FALSE

Comment (Time-Green) [reply code/datums/sprite_accessories.dm:1909]: Oh I misunderstood. Honestly just rename it to` /datum/sprite_accessory/fox_ears`. It's cringe but sprice_accessory is already hopeless imo

File: code/modules/client/preferences/species_features/lizard.dm [hunk 4]
```diff
@@ -93,13 +93,39 @@
...
+	return list(NORMAL_LEGS, DIGITIGRADE_LEGS)
 
 /datum/preference/choiced/lizard_legs/apply_to_human(mob/living/carbon/human/target, value)
 	target.dna.features["legs"] = value
+	// Hack to update the dummy in the preference menu
+	// (Because digi legs are ONLY handled on species change)
+	if(!isdummy(target) || target.dna.species.digitigrade_customization == DIGITIGRADE_NEVER)
+		return
+
+	var/list/correct_legs = target.dna.species.bodypart_overrides.Copy() & list(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
```
Comment (Time-Green) [inline code/modules/client/preferences/species_features/lizard.dm:107]: Doesn't this already make a new list, making .Copy() unneccesary?

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 5]
```diff
@@ -43,13 +42,20 @@
...
 		else
-			var/obj/item/organ/internal/ears/cat/ears = new(FALSE, target_human.dna.features["ears"])
+			var/obj/item/organ/internal/ears/cat/ears = new()
 			ears.Insert(target_human, movement_flags = DELETE_IF_REPLACED)
 	return ..()
 
 /datum/species/human/felinid/randomize_features(mob/living/carbon/human/human_mob)
 	var/list/features = ..()
-	features["ears"] = pick("None", "Cat")
+	// Future todo: Integrate this with the bodypart overlay itself better
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:51]: :/

File: code/modules/surgery/organs/internal/ears/_ears.dm [hunk 6]
```diff
@@ -15,6 +15,10 @@
 	now_fixed = "<span class='info'>Noise slowly begins filling your ears once more.</span>"
 	low_threshold_cleared = "<span class='info'>The ringing in your ears has died down.</span>"
 
+	/// What broad type of ear overlay do we have
+	var/ear_overlay_type
+	/// The ear overlay type
+	var/datum/bodypart_overlay/mutant/ear_overlay
```
Comment (Time-Green) [inline code/modules/surgery/organs/internal/ears/_ears.dm:21]: Not ideal obviously, but it'll be fixed by #85126

File: code/modules/surgery/organs/internal/ears/_ears.dm [hunk 7]
```diff
@@ -225,3 +236,46 @@
...
+
+/datum/bodypart_overlay/mutant/ears/felinid/New()
+	. = ..()
+	set_appearance(/datum/sprite_accessory/ears/cat)
+
+/datum/bodypart_overlay/mutant/ears/fox
+
+/datum/bodypart_overlay/mutant/ears/fox/New()
+	. = ..()
+	set_appearance(/datum/sprite_accessory/ears/fox)
```
Comment (Time-Green) [inline code/modules/surgery/organs/internal/ears/_ears.dm:281]: You can set the sprite_datum var and set imprint_on_next_insertion to FALSE for an override like this

**Comment by CliffracerX** [comment]
Important question about this - will this impact Genetics' ability to change people's appearances?  (I.E, being able to modify the string to modify someone's tail or ears type)

**Comment by MrMelbert** [comment]
it *shouldn't*

**Comment by MrMelbert** [comment]
i'll be back to work on this next week probably

**Pull request closed without merging**